### ISSUE TITLE
Next Iteration Meeting ??-05-2023

### ISSUE THREAD

**Issue by AbdBarho** [LAION-AI/Open-Assistant#3007]

**Comment by jjmachan** [comment]
First, have to say this is an awesome initiative to get the whole community involved kudos to you guys 🙌🏽 ❤️ 

I have two questions /topics
1. Plans to make OA sustainable in the long run?
2. What is the unique value we can provide to the wider community based on the type of people and experiences we have here? in short, what are our unique advantages that we can capitalize on to build something unique. what are we uniquely positioned to do?

Looking to catch you guys in the call :)

**Comment by RainRat** [comment]
Data cleaning website:

1. Fix code formatting, correct minor errors. Possible reranking by experts.
2. Tag every variant of "as a language model", identifying as a specific AI, or implementation detail ("cutoff date of"). Users can then choose to leave them out if they desire to give their chatbot its own personality.
3. Tag every example of refusing to do something a chatbot with a plugin can do. Users can leave them out so their plugin enabled chatbot has less chance of refusing task.

**Comment by 0x22almostEvil** [comment]
1. More context length (recent paper)
2. Multimodal (adding capability to get an information from images, audios, videos etc.)

**Comment by sedthh** [comment]
1. more info on plugins and how to contribute in this area
2. options for AutoGPT alternative
3. updates on safety and community guidelines --> are there any new decisions on firewalls, etc.?
4. requested functions from user feedback survey, which one to work on and when to expect them

**Comment by notmd** [comment]
Discuss how we gonna do with auth service. 
- Should we just implement a [SSO](https://auth0.com/blog/what-is-and-how-does-single-sign-on-work/) or just a simple cookie-sharing solution (this should be easy since we are using only one domain)
- How we gonna migrate the existing data to the new service.

**Comment by andreaskoepf** [comment]
(wip)

## Potential extensions of the project:

- plugins (examples, documentation, video)
- retrieval (access wikipedia & custom document knowledge-bases)
- multi-modal (starting with image embeddings)
- forming data collection expert groups data (natural sciences, math, law, business administration, finance, engineering, software development, philosophy, literature, art, linguistic research), similar to stack exchange
- use multiple models to answer requets (sparse-disributed, mixture of experts)
- help other communities to create own versions of OA (e.g. Chinese)

## Community projects to trigger and feature:

- voice interaction with OA

## Overall Strategy

- focus on human feedback collection and model improvements, try to deliver the best open-source conversational models
    - drop load from the system which does not generate valuable human feedback, explititly ask people to run general pupose inference on their own systems, potentially form a sub-group which is interested in offering larger scale OA inference and decouple this from main site
- transform the chat into a model-teacher UI:
    - add flag item to report inacceptable resonses
    - generate multiple replies (e.g. 3) and let users rank
    - let users edit (correct) model resonses
    - show trees of to other users for ranking or rating, especially to confirm edits
    - collect feedback about the response quality and why a reply was good/bad
- continuous integration of feedback: automate the training process, build a model factory that allows to configure the data mix and pulls and prepares data from multiple data-sources

## Inference System

- separate core inference system from human feedback collection, offering as stand-alone customizable basis (logos etc.) for others to quickly build and deploy theri chat interfaces

## Backend

- let users decide on which prompts they want to work
- implement most wanted features: prompt-response pairs,

## Model

- great results from RL fine-tuning
- fine-tuning and deployment of llama 65B

**Comment by 0x22almostEvil** [comment]
3. Add hints for prompts (like in Bing)
![image-31](https://user-images.githubusercontent.com/122345469/235766592-ede638a8-1fdf-4989-a1ad-8a1e63948e1e.png)
![bing_refuses_to_Talk](https://user-images.githubusercontent.com/122345469/235766597-0bccdd8f-be7c-4d6b-aaf9-2656aa2788bb.jpg)


> 1. More context length (recent paper)
> 2. Multimodal (adding capability to get an information from images, audios, videos etc.)

**Comment by MattAlexMiracle** [comment]
Utilize feedback from deployed assistant (needs additional considerations wrt filterting etc...)

**Comment by sedthh** [comment]
+1 usage statistics: how the number of active users / contributors changes since the release

**Comment by theblackcat102** [comment]
Expand language support to include non latin languages ( chinese, japanese, korean, thai )

Reasons:

 * our llama 13B trained on oasst with the aforementioned languages performs quite well
 
 * the chinese discord channel has been sharing how to "hack" our 30B sft model to response in chinese, so basically even without training llama model has been able to perform pretty well
 
* this would increase positive exposure in Asia pacific, thereby increase the flywheel size

**Comment by DividesZero** [comment]
The future steps towards becoming an assistant rather than just a chatGPT replacement (plugins in the near future but what else, what's in the FAR future, etc).

**Comment by 0x22almostEvil** [comment]
- External dataset legal problems

**Comment by theblackcat102** [comment]
# Open Assistant Meetup notes - 06/05/2023

Here's a list of points that we covered during the meeting:

- multi-modalities are mentioned multiple time
    - storage might be an issue
- we also need a RLHF on the results generated by human, not on human generated response
    - mainly the issue is oasst lack of bad example
- what if I want a open assistant for gardening, finance
    - a lot of people asking if we could use <narrow task> in our <some place>
    - expert LLM for a specific domain
- model size
    - model parameters is still a major factor in perf
- extensions
    - yannic will update the new tool model to hf API tomorrow
    - being able to use tool is very important
    - **context length is still a problem, 8k is needed for it to work**
    - dragan show case a vercel website deployment on the same prompts used by openai and OA models
        - performance are similar to gpt-3.5-turbo
- retrieval
    - able to simply use the embedding a fed directly into the LLM
    - basically it would be faster than placing it in the prompt
- meta prompt editing
    - andreas propose we need a prompt editing
- discord bot integration
    - with thumbs up and down
- API for open source development
    - yannic is working on the API with authentication
- sustainability
    - will be very ugly once we have to setup entities for this

- ask people to donate compute power in a decentralized manner, and reward would just be prioritized in queue for chat

**Comment by smeyerhot** [comment]
```ask people to donate compute power in a decentralized manner, and reward would just be prioritized in queue for chat```

https://github.com/bigscience-workshop/petals
